documentation/src/docs/clients-and-streams.md

```diff
@@ -55,51 +69,99 @@ ClientConfig clientConfig = ClientConfig.builder()
 
 We can read back these events using a reader. Readers are associated with reader groups, which track the readers’ progress and allow more than one reader to coordinate over which segments they’ll read.
 
-```java
-ReaderGroupManager readerGroupManager = ReaderGroupManager
+=== "Java"
+
+    ``` java
+    ReaderGroupManager readerGroupManager = ReaderGroupManager
         .withScope("tutorial", clientConfig);
-ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+    ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
         .stream("tutorial/numbers").build();
-readerGroupManager.createReaderGroup("numReader", readerGroupConfig);
-```
+    readerGroupManager.createReaderGroup("numReader", readerGroupConfig);
+    ```
+
+=== "Python"
+
+    ``` python
+    # create a ReaderGroup rg1 against an already created Pravega scope and Stream.
+    reader_group = stream_manager.create_reader_group("numReader", "tutorial", "numbers")            
+    ```
 
 Here we used the [`ReaderGroupManager`](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/admin/ReaderGroupManager.html) to create a new reader group on numbers called `numReader`.
 
 Now we can attach a single [`EventStreamReader`](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/stream/EventStreamReader.html) instance to our reader group and read our 3 numbers from our numbers stream:
 
-```java
-EventStreamReader<Integer> reader = factory
+=== "Java"
+
+    ``` java
+    EventStreamReader<Integer> reader = factory
         .createReader("myId", "numReader",
                       new JavaSerializer<Integer>(), ReaderConfig.builder().build());
 
-Integer intEvent;
-while ((intEvent = reader.readNextEvent(1000).getEvent()) != null) {
-    System.out.println(intEvent);
-}
+    EventRead<Integer> eventRead;
+    Integer intEvent = null;
+    while ((eventRead = reader.readNextEvent(1000)).isCheckpoint() || (intEvent = eventRead.getEvent()) != null) {
+        if (intEvent != null) {
+            System.out.println(intEvent);
```

Comment:
Here and elsewhere perhaps we should invoke some method like "process" so that readers are clear on what this is meant to stand in for